src/functions.rs

```diff
@@ -13,3 +29,64 @@ pub(crate) fn get_proposal_bytes(proposal: String) -> Vec<u8> {
 		return hex::decode(contents.as_str().trim_start_matches("0x")).expect("Valid proposal")
 	}
 }
+
+/// Convert a wall clock time string to a block number
+/// Format: DD-MM-YYThhmm (e.g. 25-05-21T0800)
+pub(crate) async fn wall_clock_to_block_number(
+	time_str: &str,
+	network: &str,
+) -> Result<u32, TimeConversionError> {
+	// Parse the input time string
+	let naive_dt = NaiveDateTime::parse_from_str(time_str, "%d-%m-%yT%H%M")
+		.map_err(|e| TimeConversionError(format!("Invalid time format: {}", e)))?;
+
+	// Convert to UTC
+	let local_dt = Local
+		.from_local_datetime(&naive_dt)
+		.single()
+		.ok_or_else(|| TimeConversionError("Invalid local time".to_string()))?;
+	let utc_dt = local_dt.with_timezone(&Utc);
+
+	// Get current block number and timestamp
+	let url = match network.to_lowercase().as_str() {
+		"polkadot" => "wss://polkadot-rpc.dwellir.com:443",
+		"kusama" => "wss://kusama-rpc.dwellir.com:443",
```

Comment:
Global issue, looks like Dwellir has added a `.n` to their URLs.

Reply:
Thanks. I’ve updated the URLs in [this PR](https://github.com/joepetrowski/opengov-cli/pull/46) so we can merge it directly.